### PR TITLE
Fix issue when user filters UTXOs by address query and custom tokens

### PR DESCRIPTION
**Pull request by tuliomir**
The feature to `send-tx` querying by filter address was only working for HTR tokens, and would have wrong behaviors when any of the outputs had a custom token.

### Acceptance Criteria
- A transaction must be successful with input address filter query and custom tokens, as described in #151 

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -703,19 +731,36 @@ walletRouter.post('/send-tx',
...
+            success: false,
+            error: 'No utxos available for the query filter for this amount.',
+            token: tokenUid
+          };
+          res.send(response);
+          lock.unlock(lockTypes.SEND_TX);
+          return;
+        }
+
+        for (const utxo of utxos) {
```
Comment (r4mmer) [inline src/index.js:758]: Why this change? We map `utxos` to `treatedInputs` then assign it to `inputs`, the same as we did before.

File: __tests__/test-utils.js [hunk 2]
```diff
@@ -47,6 +47,9 @@ class TestUtils {
     "Weg6WEncAEJs5qDbGUxcLTR3iycM3hrt4C",
     "WSVarF73e6UVccGwb44FvTtqFWsHQmjKCt",
     "Wc5YHn861241iLY42mFT8z1dT1UdsNWkfs",
+    "WU6KGNPuaRG4VCKC1fsKDh4fRZGbkqxG8Y",
```
Comment (r4mmer) [inline __tests__/test-utils.js:50]: Why these new addresses?

File: src/index.js [hunk 3]
```diff
@@ -563,6 +571,13 @@ const getUtxosToFillTx = (wallet, sumOutputs, options) => {
   }
 }
 
+/**
+ * @typedef TokenOutput
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/index.js:575]: This type is first declared on line 704 and used on 718:
https://github.com/HathorNetwork/hathor-wallet-headless/blob/ca9f1938522cfe8ee86e6f712e3d2224f53215a3/src/index.js#L718-L722

It is used as a helper object only for internal processes, with no interface to the clients.

File: src/index.js (hunk 1 above)
Comment (tuliomir) [reply src/index.js:758]: Each token iterated may have multiple `utxos`. The `treatedInputs` variable is basically an intermediate to reduce them all to a single resulting array.

Only after all the output tokens have been iterated we overwrite the `inputs` variable with the final result. That was the most concise solution I found.
Do you have an alternative suggestion?

File: __tests__/test-utils.js (hunk 2 above)
Comment (tuliomir) [reply __tests__/test-utils.js:50]: Since I had to use more of the wallet's addresses on the new tests, any query that involves the _gap limit_ also fetched more addresses that were not specified before on the mocked code.

For example, this `addresses` test:
https://github.com/HathorNetwork/hathor-wallet-headless/blob/ca9f1938522cfe8ee86e6f712e3d2224f53215a3/__tests__/addresses.test.js#L28

This is part of the refactoring I had to do on all the mocked tests fixtures on commit https://github.com/HathorNetwork/hathor-wallet-headless/pull/160/commits/e8f39488a88d573a63c581d6d9d8734f8fa1b08c

File: src/index.js (hunk 1 above)
Comment (r4mmer) [reply src/index.js:758]: Didn't see that the inputs assignment was on another context. 
But i guess we could remove the `treatedInputs` anyway with:
```js
      // init inputs as empty array then
      inputs = inputs.concat(utxos.map((utxo) => {
        return {txId: utxo.tx_id, index: utxo.index};
      }));
```

File: src/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/index.js:575]: I understood the usage, but shouldn't this comment be better near where it's used?
I think this would avoid confusion.

File: __tests__/integration/send-tx.test.js [hunk 4]
```diff
@@ -1359,3 +1359,246 @@ describe('send tx (custom tokens)', () => {
...
+  });
+
+  afterAll(async () => {
+    await wallet1.stop();
+  });
+
+  it('should reject for insufficient funds (custom)', async done => {
+    // Sending all the tokens to facilitate address-info validation
+    const txErr = await wallet1.sendTx({
+      inputs: [{ type: 'query', filter_address: await wallet1.getAddressAt(0) }],
```
Comment (pedroferreira1) [inline __tests__/integration/send-tx.test.js:1403]: I would just add a new error test case with amount 100 (an amount that the wallet has) but with an address without the utxo, e.g. the address 1.

**Review by pedroferreira1 — APPROVED** [review]
